orm/config-file.md

```diff
@@ -0,0 +1,316 @@
+    //LaravelDoctrine\ORM\Extensions\TablePrefix\TablePrefixExtension::class,
+    //LaravelDoctrine\Extensions\Timestamps\TimestampableExtension::class,
+    //LaravelDoctrine\Extensions\SoftDeletes\SoftDeleteableExtension::class,
+    //LaravelDoctrine\Extensions\Sluggable\SluggableExtension::class,
+    //LaravelDoctrine\Extensions\Sortable\SortableExtension::class,
+    //LaravelDoctrine\Extensions\Tree\TreeExtension::class,
+    //LaravelDoctrine\Extensions\Loggable\LoggableExtension::class,
+    //LaravelDoctrine\Extensions\Blameable\BlameableExtension::class,
+    //LaravelDoctrine\Extensions\IpTraceable\IpTraceableExtension::class,
+    //LaravelDoctrine\Extensions\Translatable\TranslatableExtension::class
+],
+```
+
+### Custom Types
+
+Custom types are classes that allow Doctrine to marshal data to/from the data source in a custom format. 
+
+To register a custom type simple add the class to this list. [For more information on custom types refer to the Doctrine documentation.](https://doctrine-orm.readthedocs.org/en/latest/cookbook/custom-mapping-types.html)
+
+### Custom Functions
+
+These are classes that extend the functionality of Doctrine's DQL language. More information on what functions are available [visit the repository.](https://github.com/beberlei/DoctrineExtensions) 
+
+To use the extensions in this sample you must install the extensions package:
+
+`require laravel-doctrine/extensions`
+
+and follow the [installation instructions.](http://www.laraveldoctrine.org/docs/1.0/extensions/installation)
+ 
+To add a function simply add it to the correct list using this format:
+
+`'FUNCTION_NAME' => 'Path\To\Class'`
+
+```php
+/*
+|--------------------------------------------------------------------------
+| DQL custom datetime functions
+|--------------------------------------------------------------------------
+*/
+'custom_datetime_functions' => [],
+/*
+|--------------------------------------------------------------------------
+| DQL custom numeric functions
+|--------------------------------------------------------------------------
+*/
+'custom_numeric_functions'  => [],
+/*
+|--------------------------------------------------------------------------
+| DQL custom string functions
+|--------------------------------------------------------------------------
+*/
+'custom_string_functions'   => [],
+```
+
+### Logger
+
+Enable logging of Laravel Doctrine and Doctrine by using the logger functionality.
+
+Available loggers:
+
+* `LaravelDoctrine\ORM\Loggers\LaravelDebugbarLogger`
+* `LaravelDoctrine\ORM\Loggers\ClockworkLogger`
+* `LaravelDoctrine\ORM\Loggers\FileLogger`
+
+` 'logger' => env('DOCTRINE_LOGGER', false),`
+
+### Cache
+
+Cache providers available:
+
+* `apc`
+* `array`
+* `file`
+* `memcached`
+* `redis`
+
+Properties:
+
+* **default** - The default cache provider to use.
+* **namespace** - 
+* **second_level** - Should second_level cache be used?
+
+### Gedmo
+
+This is an option for use with **Extensions**
```

Comment:
- Namespace 
Will add namespace to the cache key. This is useful if you need extra control over handling key names collisions in your Cache solution.

- 2nd level cache:
The Second Level Cache is designed to reduce the amount of necessary database access. It sits between your application and the database to avoid the number of database hits as much as possible.

When turned on, entities will be first searched in cache and if they are not found, a database query will be fired an then the entity result will be stored in a cache provider.

When used, READ_ONLY is mostly used. ReadOnly cache can do reads, inserts and deletes, cannot perform updates
```
/**
 * @Entity
 * @Cache(usage="READ_ONLY", region="my_entity_region")
 */
class User {
```